cirq-web/cirq_ts/src/circuit/main.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Scene, PerspectiveCamera, WebGLRenderer} from 'three';
+import {OrbitControls} from 'three/examples/jsm/controls/OrbitControls';
+import {GridCircuit} from './grid_circuit';
+import {GridCoord} from './components/types';
+
+/**
+ * Creates a three.js scene object, adds it to the container element
+ * at the designated id, and adds the orbit control functionality.
+ * @param numQubits The number of qubits in the circuit
+ * @param sceneId The container id that will host the three.js scene object
+ * @returns The three.js scene object
+ */
+function createAndRenderScene(numQubits: number, sceneId: string): Scene {
+  const WIDTH = 1000;
+  const HEIGHT = 700;
+  const NUM_QUBITS = numQubits;
+
+  const scene = new Scene();
+  const camera = new PerspectiveCamera(75, WIDTH / HEIGHT, 0.1, 1000);
+
+  const renderer = new WebGLRenderer({alpha: true});
+  const controls = new OrbitControls(camera, renderer.domElement);
+
+  renderer.setSize(WIDTH, HEIGHT);
+  const el = document.getElementById(sceneId)!;
+
+  el.appendChild(renderer.domElement);
+
+  // TODO: create a better way to start the
```

Comment:
yeah, it would be nice to see the whole circuit from a distant enough view point